packages/use-dataloader/src/dataloader.ts

```diff
@@ -1,248 +1,152 @@
+  public key: string
 
-  public keepPreviousData?: boolean
+  public method: () => PromiseType<ResultType>
 
-  private errorListeners: Array<OnErrorFn> = []
+  public isCalled = false
 
-  private successListeners: Array<OnSuccessFn<T>> = []
+  public isCancelled = true
 
-  private cancelListeners: Array<OnCancelFn> = []
+  public status: StatusEnum = StatusEnum.IDLE
 
-  private observerListeners: Array<(dataloader: DataLoader<T>) => void> = []
+  public error?: ErrorType
 
-  public error?: Error
+  public data?: ResultType
 
-  private dataOutdatedTimeout?: number
+  public observers: Array<() => void> = []
 
   public timeout?: number
 
-  private destroyed = false
+  public loadCount = 0
+
+  public isFirstLoading = true
 
-  public constructor(args: DataLoaderConstructorArgs<T>) {
+  public constructor(args: DataLoaderConstructorArgs<ResultType>) {
     this.key = args.key
-    this.status = StatusEnum.IDLE
     this.method = args.method
-    this.pollingInterval = args?.pollingInterval
-    this.keepPreviousData = args?.keepPreviousData
-    this.maxDataLifetime = args.maxDataLifetime
-    this.needPolling = args.needPolling ?? true
-    this.notifyChanges()
+    if (args.enabled) {
+      this.status = StatusEnum.LOADING
+    }
+    this.data = DataLoader.cachedData[this.key] as ResultType
+    if (args.notifyChanges) this.observers.push(args.notifyChanges)
   }
 
-  public getData(): T | undefined {
-    return (DataLoader.cachedData[this.key] as T) ?? undefined
+  public notifyChanges() {
+    if (this.timeout) {
+      clearTimeout(this.timeout)
+    }
+    this.timeout = setTimeout(() => {
+      this.observers.forEach(observer => observer())
+    }) as unknown as number
   }
 
-  private notifyChanges(): void {
-    this.observerListeners.forEach(observerListener => observerListener(this))
+  public getData(): ResultType | undefined {
+    return DataLoader.cachedData[this.key] as ResultType
   }
 
-  public load = async (force = false): Promise<void> => {
-    if (
-      force ||
-      this.status === StatusEnum.IDLE ||
-      this.status === StatusEnum.ERROR ||
-      (this.status === StatusEnum.SUCCESS && this.isDataOutdated)
-    ) {
-      if (DataLoader.started < DataLoader.maxConcurrent) {
-        // Because we want to launch the request directly without waiting the return
-        if (this.timeout) {
-          // Prevent multiple call at the same time
-          clearTimeout(this.timeout)
-        }
-        await this.launch()
-      } else {
-        await new Promise(resolve => {
-          setTimeout(resolve)
-        }).then(() => this.load(force))
-      }
+  private tryLaunch = async (): Promise<ResultType> => {
+    if (DataLoader.started < DataLoader.maxConcurrent) {
+      DataLoader.started += 1
+      DataLoader.queue[this.key] = this.launch()
+    } else {
+      DataLoader.queue[this.key] = new Promise(resolve => {
+        setTimeout(resolve)
+      }).then(() => this.tryLaunch())
```

Comment:
Not sure if it's easier or not to read
```suggestion
      }).then(this.tryLaunch)
```